### PR TITLE
Lint files in `lib/librarian` directory.

### PR DESCRIPTION
**Pull request by tzmanics**
Ran a few test but will test completely after review.

### CODE REVIEW

File: lib/librarian/http.js [hunk 1]
```diff
@@ -35,55 +36,45 @@ var HTTP = function(host, port, ssl) {
...
 
-  var req = this._corehttp.request(options, function(res){
+  req = this._corehttp.request(options, function (res) {
     callback(null, req, res);
   });
 
-  req.on('error', function(err){
-    console.log(err);
+  req.on('error', function (err) {
     callback(err, req, null);
```
Comment (franvarney) [inline lib/librarian/http.js]: Remove `null`

Comment (franvarney) [inline lib/librarian/http.js]: Remove `null`

File: lib/librarian/http.js [hunk 2]
```diff
@@ -35,55 +36,45 @@ var HTTP = function(host, port, ssl) {
...
+  // Data is optional. If parameter is a function, assume it's the callback.
+  if (typeof data === 'function') {
     callback = data;
     data = null;
   }
 
   // Convert the incoming data to JSON.
-  var headers = null;
-  if(data) {
+  headers = null;
```
Comment (franvarney) [inline lib/librarian/http.js]: Unnecessary

**Comment by franvarney** [comment]
:+1:

File: lib/librarian/http.js [hunk 3]
```diff
@@ -35,55 +36,45 @@ var HTTP = function(host, port, ssl) {
...
+  var req, options;
+  if (typeof data === 'function') {
     callback = data;
     data = null;
   }
 
-  var headers = null;
-  if(data) {
-    data = JSON.stringify(data);
-    headers = {
```
Comment (jackboberg) [inline lib/librarian/http.js]: Seems like someone probably intended for this to be added to the options. Maybe leave a `FIXME/TODO` in comments?

**Comment by tzmanics** [comment]
rebased with linting branch and added noted fixes.

**Pull request merged**